src/cryptography/x509.py

```diff
@@ -1344,3 +1344,52 @@ def extensions(self):
         """
         Returns an Extensions object containing a list of Revoked extensions.
         """
+
+
+class CertificateSigningRequestBuilder(object):
+    def __init__(self, version=Version.v1, subject_name=None, extensions=[]):
+        """
+        Creates an empty X.509 certificate request (v1).
+        """
+        self._version = Version.v1
+        self._subject_name = subject_name
+        self._extensions = extensions[:]
+
+    def set_version(self, version):
```

Comment:
We don't parse CSR version at all right now which is how that is being ignored. I'm not 100% sure what we should do here, but to my knowledge there's no actual 0x2 version for CSRs. RFC 2986 defines v1.7 of PKCS#10 (unless something obsoletes it, which I have been unable to find) and only defines a v1. In practice since it's an ASN.1 integer you can obviously put whatever you want there.